File: wear/src/main/java/at/shockbytes/corey/dagger/WearAppComponent.java
package at.shockbytes.corey.dagger;

import javax.inject.Singleton;

import at.shockbytes.corey.core.MainActivity;
import at.shockbytes.corey.fragment.MainFragment;
import at.shockbytes.corey.fragment.WorkoutFragment;
import dagger.Component;

/**
 * @author Martin Macheiner
 *         Date: 21.02.2017.
 */
@Singleton
@Component(modules = {WearAppModule.class})
public interface WearAppComponent {

	void inject(MainActivity activity);

	void inject(MainFragment fragment);

    void inject(WorkoutFragment fragment);

}

File: wear/src/main/java/at/shockbytes/corey/dagger/WearAppComponent.kt
package at.shockbytes.corey.dagger

import at.shockbytes.corey.ui.activity.WearMainActivity
import at.shockbytes.corey.ui.fragment.WearRunningFragment
import at.shockbytes.corey.ui.fragment.WorkoutFragment
import at.shockbytes.corey.ui.fragment.workoutpager.WearTimeExercisePagerFragment
import dagger.Component
import javax.inject.Singleton

/**
 * Author:  Martin Macheiner
 * Date:    21.02.2017
 */
@Singleton
@Component(modules = [(WearAppModule::class), (WearViewModelModule::class)])
interface WearAppComponent {

    fun inject(activity: WearMainActivity)

    fun inject(fragment: WorkoutFragment)

    fun inject(fragment: WearRunningFragment)

    fun inject(fragment: WearTimeExercisePagerFragment)
}
